### PR TITLE
Update function syntax from `λx.{...}` to `fun x {...}`

### PR DESCRIPTION
**Pull request by minchingtonak**
Pretty self-explanatory, all necessary features are implemented.

The only other thing I want to do is move common textual representations for Hazel constructs to their own file so that there aren't duplicated magic string literals at a couple points in the code base. I've done it just for function syntax (see `Doc_common.re`, `DHDoc_common.re`, `UHDoc_common.re` for defs and usage) but I think it would be a good idea to move all of the syntax over to `Doc_common.re`. That's probably best left for a separate PR, but please let me know if you think I should include it in this PR.

### CODE REVIEW

File: src/hazelcore/Action_Exp.re [hunk 1]
```diff
@@ -74,6 +74,7 @@ let keyword_action = (kw: ExpandingKeyword.t): Action.t =>
   switch (kw) {
   | Let => Construct(SLet)
   | Case => Construct(SCase)
+  | Lam => Construct(SLam)
```
Comment (cyrus-) [inline src/hazelcore/Action_Exp.re]: Let's call it "Fun" instead of "Lam" given the concrete syntax

File: src/hazelweb/layout/Doc_common.re [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module Delim = {
+  let sym_Lam = "fun";
```
Comment (cyrus-) [inline src/hazelweb/layout/Doc_common.re]: Fun here too

**Review by cyrus- — CHANGES_REQUESTED** [review]
looking close, but needs to be brought up to date with dev in a big way

**Comment by cyrus-** [comment]
@minchingtonak do you want to try to wrap this up? looks quite close -- just a few changes above, plus merging in the latest `dev` changes.

**Comment by minchingtonak** [comment]
Yep, thanks for the reminder! I'll start this today.

**Comment by minchingtonak** [comment]
I refactored Lam to Fun in ExpandingKeyword.t and Action.t. I'm not sure whether you want me to change Action.t so I can rool that back if need be. I also changed the string constants from Lam -> Fun in Doc, DHDoc, and UHDoc files.

**Review by cyrus- — APPROVED** [review]

**Review by cyrus- — CHANGES_REQUESTED** [review]
Whoops, one last late-breaking change needed now that we have Hannah's work on the strategy guide working -- the entries for creating function literals should say `fun ` in addition to `\`.

**Comment by cyrus-** [comment]
Screenshot from 2021-11-05 13-36-06

**Review by cyrus- — CHANGES_REQUESTED** [review]
Looks great. One last issue: the undo history leaves a "insert var: fun" entry after pressing space, whereas with other keywords like "let" and "case" it overrides the previous entry.

File: src/hazelcore/Statics_Exp.re [hunk 3]
```diff
@@ -768,9 +768,10 @@ and syn_fix_holes_operand =
       | NotInVarHole =>
         let (u, u_gen) = MetaVarGen.next(u_gen);
         let reason: VarErrStatus.HoleReason.t =
-          switch (Var.is_let(x), Var.is_case(x)) {
-          | (true, _) => Keyword(Let)
-          | (_, true) => Keyword(Case)
+          switch (Var.is_let(x), Var.is_case(x), Var.is_fun(x)) {
```
Comment (cyrus-) [inline src/hazelcore/Statics_Exp.re]: Lets use ExpandingKeyword.mk here so there is only one place we need to modify when there is a new keyword added to the system

File: src/hazelcore/Var.re [hunk 4]
```diff
@@ -25,10 +25,14 @@ let is_false = eq("false");
 
 let is_let = eq("let");
 
+let is_fun = eq("fun");
+
 let is_case = eq("case");
 
 let is_wild = eq("_");
 
+let is_keyword = s => is_let(s) || is_fun(s) || is_case(s);
```
Comment (cyrus-) [inline src/hazelcore/Var.re]: Let's change this to use ExpandingKeyword.mk -- same goal, to have it so we don't need to find multiple places in the codebase to change when a new keyword is added

File: src/hazelweb/layout/DHDoc_Util.re [hunk 5]
```diff
@@ -58,10 +58,10 @@ module Delim = {
   let open_Parenthesized = mk("(");
   let close_Parenthesized = mk(")");
 
-  let sym_Lam = mk(Unicode.lamSym);
-  let colon_Lam = mk(":");
-  let open_Lam = mk(".{");
-  let close_Lam = mk("}");
+  let sym_Fun = mk(Unicode.lamSym);
+  let colon_Fun = mk(":");
+  let open_Fun = mk(".{");
```
Comment (cyrus-) [inline src/hazelweb/layout/DHDoc_Util.re]: this is inconsistent with the UHExp pretty printing -- no dot there

**Review by cyrus- — CHANGES_REQUESTED** [review]
Just a few last code cleanups and this looks good to go!

**Review by cyrus- — APPROVED** [review]
Looks great!

**Pull request merged**